modules/transports/core/nhttp/src/main/java/org/apache/synapse/transport/passthru/core/MultiListenerServerIODispatch.java

```diff
@@ -46,32 +49,59 @@ public MultiListenerServerIODispatch(final Map<Integer, NHttpServerEventHandler>
     protected void onConnected(final DefaultNHttpServerConnection defaultNHttpServerConnection) {
         int localPort = defaultNHttpServerConnection.getLocalPort();
         NHttpServerEventHandler handler = handlers.get(localPort);
-        try {
-            handler.connected(defaultNHttpServerConnection);
-        } catch (final Exception ex) {
-            handler.exception(defaultNHttpServerConnection, ex);
+        if (handler == null) {
+            log.error("Listener has not started yet on " + localPort + ". Hence shutting down the connection");
```

Comment:
Shall we change the error log to something like, "The request to the port + localPort + cannot be processed further as the Listener has not yet started on port" + localPort?

Reply:
Done